pkg/oci/dockerkeychain/docker_keychain.go

```diff
@@ -20,6 +21,54 @@ type DockerKeychain struct {
 	mutex        sync.Mutex
 }
 
+func NewDockerKeychains(ctx context.Context, apiReader client.Reader, namespaceName string, pullSecretsNames []string) (authn.Keychain, error) {
+	keychain := &DockerKeychain{}
+	err := keychain.loadDockerConfigFromSecrets(ctx, apiReader, namespaceName, pullSecretsNames)
+
+	return keychain, err
+}
+
+func (keychain *DockerKeychain) loadDockerConfigFromSecrets(ctx context.Context, apiReader client.Reader, namespaceName string, pullSecretsNames []string) error {
+	if len(pullSecretsNames) == 0 {
+		return nil
+	}
+
```

Comment:
You don't need two functions here; you can put them together in a single one (only `NewDockerKeychains` is referenced across the code)

Reply:
NewDockerKeychain is used by `troubleshoot` command

Reply:
That's what I mean! You can remove `loadDockerConfigFromSecrets()` and put its implementation inside `NewDockerKeychains()`